src/epcpm/sunspectoxlsx.py

```diff
@@ -484,8 +492,12 @@ def gen(self):
             row.description = parameter.comment
             row.read_write = "R" if parameter.read_only else "RW"
             row.parameter_uuid = parameter.uuid
-            # TODO: Unclear why the interfaceItem_<UUID> is not defined for DataPointBitfield.
             row.parameter_uses_interface_item = False
+            row.type_uuid = self.wrapped.type_uuid
+            # TODO: Unclear whether not_implemented==True is correct for bitfields?
+            row.not_implemented = True
```

Comment:
To verify what you are saying, in the SunSpec model, there are data points of type `bitfield16`/`bitfield32` that have the `not_implemented` field. There are several examples in the SunSpec model of implemented and not implemented data points (class object `DataPoint`) of type `bitfield16`/`bitfield32`. It's not immediately clear if the not implemented data points return the proper value depending on the type? I see in `embedded-library/system/sunspec/sunspec.h` defines like this:
`#define SUNSPEC_BITFIELD32_NOT_IMPLEMENTED (0xFFFFFFFFUL)`
but they don't appear to be used anywhere? Also, I don't see a corresponding define for `bitfield16` (`SUNSPEC_BITFIELD16_NOT_IMPLEMENTED`)? I imagine checking if that functionality works is outside the scope of this PR.

What I also see are three examples of `DataPointBitfield` objects and their associated child `DataPointBitfieldMember` objects. Neither type of object has the `not_implemented` field, which was the basis for my TODO comment. If I read what you say correctly, the `DataPointBitfield` class should define the `not_implemented` field and that you would expect the value to be "`False`" for the three examples? I would also assume that the `DataPointBitfieldMember` class does not need the `not_implemented` field as it would be provided by the parent `DataPointBitfield`?

So far, I've only discussed how this applies to SunSpec. For static modbus, I suppose the answer will be clear when I verify that the SunSpec side is handled as I describe above?

Reply:
Let's go with "yes".

We probably have some regular `DataPoint`s with type `bitfield16`/`32` and some of them may be not implemented.  But, we shouldn't (presently) have any `DataPointBitfield`s that are 'not implemented'.  Whether you add `not_implemented` to `DataPointBitfield` at this point doesn't really matter to me.  But yes, the value there (and here) should be `False`.

I know we have some code that fills in values for the not implemented values.  There may be a problem with it, or maybe not.

Not implemented for static modbus might relate to the comments I've made about what happens when we do decide to EOL a point.  Also, at least returning the values that presently are returned might be important if anyone has partially implemented tidbits of SunSpec and expects static Modbus to respond in the same way for those points.

Reply:
I set `not_implemented` to False for the three bitfield examples. I think the change suffices for the meantime? See 7b4d874c64ffb3e7af342ef73218e4c7a491d14b. More work can be done on bitfield in a future PR.